src/main/webapp/script.js

```diff
@@ -13,7 +13,58 @@
 // limitations under the License.
 
 function addThreads() {  // eslint-disable-line
-  fetch('/data').then((response) => response.text()).then((quote) => {
-    document.getElementById('thread-container').innerText = quote;
+  const url = '/data';
+  fetch(url).then((response) => response.json()).then((threadInfoList) => {
+    const threadList = document.getElementById('thread-container');
+    threadList.innerHTML = '';
+
+    let i;
+    for (i = 0; i < threadInfoList.length; i += 4) {
+      // create description
+      const ulid = 'ul' + i;
+      const lidescription = document.createElement('li');
+      lidescription.innerText = 'Description';
+      lidescription.className = 'description-li';
+      const description = document.createElement('ul');
+      description.appendChild(lidescription);
+      description.appendChild(
+          createListElement('sentiment:' + threadInfoList[i + 1]));
+      description.appendChild(
+          createListElement('upvotes: ' + threadInfoList[i + 2]));
+      description.appendChild(linkListElement(threadInfoList[i + 3]));
+      description.className = ('description');
+      description.setAttribute('id', ulid);
```

Comment:
What do you think of extracting some of this to another function (which would take care of creating the item to add to the list)? It would help make the body of the for loop more concise and readable

Reply:
I could do that in the next PR along with creating the json object for the thread data. I think with the addition of the json object it will become even easier for readability within the script as well.

Reply:
Sounds good!